syntax = "proto3";
import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";

message  ForwardSensorMeteoData {
  int32 id = 1;
  message RawMeteoData {
    float temperature = 1;
    float humidity = 2;
    google.protobuf.Timestamp timestamp = 3;
  }
}

message ForwardSensorPollutionData {
  int32 id = 1;
  message RawPollutionData {
    float co2 = 1;
    google.protobuf.Timestamp timestamp = 2;
  }
}


service Server {
  rpc processMeteoData (ForwardSensorMeteoData) returns (google.protobuf.Empty) {}
  rpc processPollutionData (ForwardSensorPollutionData) returns (google.protobuf.Empty) {}
}